src/ensembl/src/content/app/entity-viewer/state/gene-view/entityViewerGeneViewActions.ts

```diff
@@ -0,0 +1,127 @@
+import { createAction } from 'typesafe-actions';
+import { ActionCreator, Action } from 'redux';
+import { ThunkAction } from 'redux-thunk';
+
+import {
+  getEntityViewerActiveGenomeId,
+  getEntityViewerActiveEnsObjectId
+} from 'src/content/app/entity-viewer/state/general/entityViewerGeneralSelectors';
+import {
+  EntityViewerGeneViewUIStateState,
+  GeneViewTabName,
+  GeneFunctionTabName,
+  GeneRelationshipsTabName,
+  EntityViewerGeneFunctionState,
+  EntityViewerGeneRelationshipsState
+} from 'src/content/app/entity-viewer/state/gene-view/entityViewerGeneViewState';
+
+import { RootState } from 'src/store';
+
+export const updateActiveGeneViewObjectState = createAction(
+  'entity-viewer/update-active-gene-view-object-state'
+)<{
+  activeGenomeId: string;
+  activeObjectId: string;
+  fragment: Partial<EntityViewerGeneViewUIStateState>;
+}>();
+
+export const setActiveGeneTab: ActionCreator<ThunkAction<
+  void,
+  any,
+  null,
+  Action<string>
+>> = (selectedTabName: GeneViewTabName) => (
+  dispatch,
+  getState: () => RootState
+) => {
+  const activeGenomeId = getEntityViewerActiveGenomeId(getState());
+  const activeObjectId = getEntityViewerActiveEnsObjectId(getState());
+
+  if (!activeGenomeId || !activeObjectId) {
+    return;
+  }
+
+  dispatch(
+    updateActiveGeneViewObjectState({
+      activeGenomeId,
+      activeObjectId,
+      fragment: {
+        selectedGeneTabName: selectedTabName
+      }
+    })
+  );
+};
+
+export const updateActiveGeneViewObjectGeneFunctionState = createAction(
+  'entity-viewer/update-active-gene-view-object-gene-function-state'
+)<{
+  activeGenomeId: string;
+  activeObjectId: string;
+  fragment: Partial<EntityViewerGeneFunctionState>;
+}>();
+
+export const setActiveGeneFunctionTab: ActionCreator<ThunkAction<
+  void,
+  any,
+  null,
+  Action<string>
+>> = (selectedTabName: GeneFunctionTabName) => (
+  dispatch,
+  getState: () => RootState
+) => {
+  const activeGenomeId = getEntityViewerActiveGenomeId(getState());
+  const activeObjectId = getEntityViewerActiveEnsObjectId(getState());
+
+  if (!activeGenomeId || !activeObjectId) {
```

Comment:
We can also add the `dispatch` statement inside the `if` clause by changing the condition to `(activeGenomeId && activeObjectId)`.

Reply:
At the moment it would be easier to understand that this function does nothing if either `activeGenomeId` or `activeObjectId` is missing. There is nothing complex going on here that has made it harder to understand anyways.

Reply:
Negated conditions are considered taboo by some who usually say to test for only `true` and not `false`. However, I get your point about letting someone reading the code to know on first glance that the function does nothing without the two ID values. What do you think about this @azangru?

Reply:
> Negated conditions are considered taboo by some who usually say to test for only true and not false.

We have definitely not been following this advice in our codebase 😄 I don't think testing for nulls, undefined or falses has impacted readability of our code.

As for whether to prefer:

```
if (condition_not_met) {
  return
}

do something
```

vs

```
if (condition_met) {
  do something
}
```

I personally would probably have written it in the second form (as more natural), but do not particularly object to reading the first one either.

Reply:
I've updated it to:
```
if (condition_met) {
  do something
}
```

I guess we should strictly follow this way going forward in order to avoid this confusion again?

Reply:
Maybe we should, but we'll all need to agree to this 😉What do you think @azangru?

Reply:
I would prefer to allow both but if we strictly need to follow one way, we may have to consider the following scenario as well:
https://github.com/Ensembl/ensembl-client/blob/efaf4e8dd434381b8024b515c52d3e77abece7fb/src/ensembl/src/content/app/entity-viewer/gene-view/components/gene-view-sidebar-tabs/GeneViewSidebarTabs.tsx#L27